### PR TITLE
Support delta & size thresholds for database_size

### PR DESCRIPTION
**Pull request by ioguix**
This pull request tries to implement what has been discussed in issue #98.

### CODE REVIEW

**Comment by Krysztophe** [comment]
@ioguix: any chance that you update this? Or shall I try? The easiest fix is probably to make critical and warning compulsory.

**Comment by ioguix** [comment]
Fixed in my branch. @Krysztophe, let me know if it is ok for you! thanks,

**Comment by Krysztophe** [comment]
I hope to test this before next year.

**Comment by ioguix** [comment]
As discussed, your tests were good enough and this only bug has been fixed and tested on my side. So I merged the branch.

Thanks!

File: check_pgactivity [hunk 1]
```diff
@@ -3542,22 +3598,58 @@ sub check_database_size {
...
+            my $c_limit;
+
+            # Check size
+            if ( defined $crit{'size'} ) {
+                $c_limit = get_size( $crit{'size'},  $db->[1] );
+                push @msg_crit => sprintf( "%s (size: %s)", $db->[0], to_size($db->[1]) )
+                    if $db->[1] >= $c_limit;
+            }
+            if ( defined $warn{'size'}
+                 and defined $c_limit and $db->[1] < $c_limit
```
[the remark this answers is not in the document]
Comment (Krysztophe) [reply check_pgactivity]: With the last changes, it is compulsory to have `delta` for both warning & critical, and|or `size` for both. A bit strict but easier to deal with, and that solves this problem.

File: check_pgactivity [hunk 2]
```diff
@@ -3542,22 +3598,58 @@ sub check_database_size {
...
+            $c_limit = undef;
+            $w_limit = undef;
+            if ( defined $crit{'delta'} ) {
+                $limit = get_size( $crit{'delta'},  $db->[1] );
+                push @msg_crit => sprintf( "%s (delta: %s)", $db->[0], to_size($delta) )
+                    if abs($delta) >= $limit;
+                $c_limit = "-$limit:$limit";
+            }
+            if ( defined $warn{'delta'}
+                 and defined $c_limit and abs($delta) < $limit
```
[the remark this answers is not in the document]
Comment (Krysztophe) [reply check_pgactivity]: same problem, same comment as above

**Comment by Krysztophe** [comment]
I was just making a last check while you were merging...

These may be problems:

  * fails to detect that critical size < warning size (that could be a feature to avoid any critical, as both thresholds are compulsory) ;
  * delta is computed as an absolute value, but it is not specified in the comments
  * delta does not accept negative values, that could be useful but was not discussed before.

And I've just found that '100B' is not a valid size value but this is more general (theoretical?) problem.

**Comment by ioguix** [comment]
> I was just making a last check while you were merging...

oops, sorry :(

> - fails to detect that critical size < warning size (that could be a feature to avoid any critical, as both thresholds are compulsory) ;

It seems to me this is a common problem with all other services. so I choose to ignore this for this PR.

> - delta is computed as an absolute value, but it is not specified in the comments

I don't understand what you mean

> - delta does not accept negative values, that could be useful but was not discussed before.

No need to allow negative value, we compare the absolute value of the database size variation with the given threshold: `if abs($delta) >= $limit`

> And I've just found that '100B' is not a valid size value but this is more general (theoretical?) problem.

Could you file a bug report with details please?

**Pull request merged**